src/component/panels/SummaryPanel/CorrelationTable/Utilities.js

```diff
@@ -0,0 +1,275 @@
+import lodash from 'lodash';
+
+const addToExperiments = (
+  experiments,
+  experimentsType,
+  type,
+  checkAtomType,
+  experimentKey,
+) => {
+  const _experiments = lodash
+    .get(experiments, `${type}`, []) // don't consider DEPT etc. here
+    .filter((_experiment) => {
+      const hasValues =
+        lodash.get(
+          _experiment,
+          type.includes('1D') ? 'ranges.values' : 'zones.values',
+          [],
+        ).length > 0;
+      return checkAtomType === true
+        ? getAtomType(_experiment.info.nucleus) === experimentKey && hasValues
+        : hasValues;
+    });
+
+  if (_experiments.length > 0) {
+    experimentsType[experimentKey] = _experiments;
+  }
+  return _experiments;
+};
+
+const getAtomType = (nucleus) => nucleus.split(/\d+/)[1];
+
+const getLabel = (correlation) => {
+  let label = Object.keys(lodash.get(correlation, 'attached', {}))
+    .map((otherAtomType) =>
+      correlation.attached[otherAtomType]
+        .map((index) => correlation.label[`${otherAtomType}${index + 1}`])
+        .filter((_label) => _label),
+    )
+    .flat()
+    .filter((_label, i, a) => a.indexOf(_label) === i)
+    .sort((a, b) =>
+      Number(a.split(/[a-z]+/i)[1]) - Number(b.split(/[a-z]+/i)[1]) < 0
+        ? -1
+        : Number(a.split(/[a-z]+/i)[1]) - Number(b.split(/[a-z]+/i)[1]) === 0 &&
+          a.split(/\d+/)[1] < b.split(/\d+/)[1]
+        ? -1
+        : 1,
+    )
+    .join('/');
+
+  if (label.length > 0) {
+    return label;
+  }
+
+  return correlation.label.origin;
+};
+
+const getLabelStyle = (state, correlation) => {
+  if (lodash.get(state, `atomType.${correlation.atomTypes[0]}.error`, false)) {
+    const error = state.atomType[correlation.atomTypes[0]].error;
+    if (
+      lodash
+        .get(error, 'outOfLimit', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'red' };
+    }
+    if (
+      lodash
+        .get(error, 'notAttached', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'blue' };
+    }
+    if (
+      lodash
+        .get(error, 'ambiguousAttachment', [])
+        .some((index) => index === correlation.index)
+    ) {
+      return { color: 'orange' };
+    }
+  }
+
+  return undefined;
+};
```

Comment:
I refactor this code for as an example of what you can do to write code better
{
const error =  lodash.get(state, `atomType.${correlation.atomTypes[0]}.error`, null) 
   if(error){
     const keys = [{key:"outOfLimit",color:"blue"},...........];
      for( let item of keys){
        if(Array.isArray(error[item .key]) && correlation.index in error[item .key]){
         return {color:item.color}; 
        }
       }
     return null;
    }